In [329]:
import pandas as pd 
import numpy as np
import matplotlib as plt

In [330]:
apps=pd.read_csv("Data_CSV/aug_train.csv") #Read the data as it is 

In [331]:
apps=apps.sort_values(by="enrollee_id") #sort the data by enrollee_id

In [332]:
#apps.to_csv('sorted_data_byID.csv', index=False) this function should run once to sort the data by ID then save CSV as sorted_data_byID

In [333]:
apps=pd.read_csv("Data_CSV/sorted_data_byID.csv") #Read the sorted data
clean_apps=apps #Create a copy of apps dataframe to save clean data

In [334]:
clean_apps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19158 entries, 0 to 19157
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   enrollee_id             19158 non-null  int64  
 1   city                    19158 non-null  object 
 2   city_development_index  19158 non-null  float64
 3   gender                  14650 non-null  object 
 4   relevent_experience     19158 non-null  object 
 5   enrolled_university     18772 non-null  object 
 6   education_level         18698 non-null  object 
 7   major_discipline        16345 non-null  object 
 8   experience              19093 non-null  object 
 9   company_size            13220 non-null  object 
 10  company_type            13018 non-null  object 
 11  last_new_job            18735 non-null  object 
 12  training_hours          19158 non-null  int64  
 13  target                  19158 non-null  int64  
dtypes: float64(1), int64(3), object(10)
me

In [335]:
missing_id=apps["enrollee_id"].isnull()
apps[missing_id]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No Missing ID</span>

In [336]:
missing_city=apps["city"].isnull()
apps[missing_city]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No Missing City</span>

In [337]:
missing_cdi=apps["city_development_index"].isnull()
apps[missing_cdi]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No Missing CDI</span>

In [338]:
missing_gender=apps["gender"].isnull()
apps[missing_gender]

enrollee_id      city  city_development_index gender  \
10              17   city_21                   0.624    NaN   
12              19   city_23                   0.899    NaN   
20              39   city_16                   0.910    NaN   
24              46   city_16                   0.910    NaN   
25              47   city_16                   0.910    NaN   
...            ...       ...                     ...    ...   
19104        33297  city_126                   0.479    NaN   
19127        33332   city_12                   0.640    NaN   
19132        33338  city_103                   0.920    NaN   
19137        33344   city_19                   0.682    NaN   
19149        33368   city_61                   0.913    NaN   

           relevent_experience enrolled_university education_level  \
10     Has relevent experience       no_enrollment        Graduate   
12     Has relevent experience    Part time course        Graduate   
20     Has relevent experience       no_enrollment         Masters   
24     Has relevent experience       no_enrollment        Graduate   
25     Has relevent experience       no_enrollment        Graduate   
...                        ...                 ...             ...   
19104  Has relevent experience    Full time course             NaN   
19127  Has relevent experience       no_enrollment        Graduate   
19132   No relevent experience    Part time course             NaN   
19137  Has relevent experience       no_enrollment        Graduate   
19149  Has relevent experience       no_enrollment        Graduate   

      major_discipline experience company_size    company_type last_new_job  \
10                STEM         13    5000-9999         Pvt Ltd            4   
12                STEM          5        50-99   Public Sector           >4   
20               Other         10      100-500         Pvt Ltd            1   
24                STEM          6        50-99   Public Sector            2   
25                STEM          5        50-99  Funded Startup            1   
...                ...        ...          ...             ...          ...   
19104              NaN        >20          <10   Public Sector           >4   
19127             STEM         11      100-500  Funded Startup            2   
19132              NaN          6          NaN             NaN            1   
19137             STEM          6          NaN             NaN            1   
19149             STEM         14      100-500         Pvt Ltd            3   

       training_hours  target  
10                 38       0  
12                 46       0  
20                 43       0  
24                  4       0  
25                 14       0  
...               ...     ...  
19104              19       1  
19127             176       0  
19132              98       0  
19137              53       1  
19149             154       0  

[4508 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 4508 Rows with missing gender</span>

In [339]:
clean_apps['gender']=apps['gender'].fillna('prefer not to say')

<span style="font-size: 20px; color: lightblue;">Result: Missing gender filled with "Prefer not to say"</span>

In [340]:
missing_exp=apps["relevent_experience"].isnull()
apps[missing_exp]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No Missing Experience</span>

In [341]:
missing_enroll=apps["enrolled_university"].isnull()
apps[missing_enroll]

enrollee_id      city  city_development_index             gender  \
71             122   city_65                   0.802               Male   
281            514  city_114                   0.926             Female   
326            604   city_16                   0.910               Male   
849           1587  city_100                   0.887               Male   
866           1613   city_72                   0.795  prefer not to say   
...            ...       ...                     ...                ...   
19017        33172   city_21                   0.624               Male   
19031        33191   city_21                   0.624  prefer not to say   
19050        33215   city_46                   0.762               Male   
19067        33241  city_115                   0.789  prefer not to say   
19096        33281   city_21                   0.624  prefer not to say   

           relevent_experience enrolled_university education_level  \
71      No relevent experience                 NaN     High School   
281    Has relevent experience                 NaN        Graduate   
326     No relevent experience                 NaN             NaN   
849    Has relevent experience                 NaN             NaN   
866    Has relevent experience                 NaN        Graduate   
...                        ...                 ...             ...   
19017  Has relevent experience                 NaN        Graduate   
19031  Has relevent experience                 NaN        Graduate   
19050   No relevent experience                 NaN        Graduate   
19067   No relevent experience                 NaN        Graduate   
19096  Has relevent experience                 NaN        Graduate   

      major_discipline experience company_size company_type last_new_job  \
71                 NaN          3          NaN      Pvt Ltd        never   
281               STEM         17          NaN          NaN           >4   
326                NaN          3          NaN          NaN        never   
849                NaN          8          NaN          NaN            1   
866               STEM         10          NaN      Pvt Ltd        never   
...                ...        ...          ...          ...          ...   
19017             STEM          7          NaN      Pvt Ltd            2   
19031             STEM          2          NaN      Pvt Ltd        never   
19050             STEM          4          NaN          NaN            1   
19067  Business Degree         <1          NaN      Pvt Ltd        never   
19096             STEM          4      100-500      Pvt Ltd            4   

       training_hours  target  
71                102       0  
281               131       0  
326                24       0  
849                62       0  
866                54       0  
...               ...     ...  
19017              65       0  
19031               8       0  
19050              44       1  
19067              52       1  
19096              63       0  

[386 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 386 Rows with missing enrolled university </span>

In [342]:
clean_apps['enrolled_university']=apps['enrolled_university'].fillna('no_enrollment')

<span style="font-size: 20px; color: lightblue;">Result: Missing enrolled university filled with "no_enrollment"</span>

In [343]:
missing_edulvl=apps["education_level"].isnull()
apps[missing_edulvl]

enrollee_id      city  city_development_index             gender  \
19              36   city_10                   0.895               Male   
65             114   city_16                   0.910               Male   
77             133  city_103                   0.920  prefer not to say   
228            420  city_103                   0.920               Male   
229            421   city_67                   0.855               Male   
...            ...       ...                     ...                ...   
19086        33265  city_114                   0.926               Male   
19097        33282   city_21                   0.624               Male   
19104        33297  city_126                   0.479  prefer not to say   
19132        33338  city_103                   0.920  prefer not to say   
19152        33374   city_61                   0.913               Male   

           relevent_experience enrolled_university education_level  \
19      No relevent experience       no_enrollment             NaN   
65     Has relevent experience       no_enrollment             NaN   
77      No relevent experience       no_enrollment             NaN   
228     No relevent experience       no_enrollment             NaN   
229    Has relevent experience       no_enrollment             NaN   
...                        ...                 ...             ...   
19086   No relevent experience    Full time course             NaN   
19097  Has relevent experience       no_enrollment             NaN   
19104  Has relevent experience    Full time course             NaN   
19132   No relevent experience    Part time course             NaN   
19152  Has relevent experience       no_enrollment             NaN   

      major_discipline experience company_size    company_type last_new_job  \
19                 NaN          3          <10         Pvt Ltd            2   
65                 NaN        >20        50-99  Funded Startup            1   
77                 NaN          5          NaN         Pvt Ltd        never   
228                NaN          7          NaN         Pvt Ltd        never   
229                NaN         14        50-99         Pvt Ltd           >4   
...                ...        ...          ...             ...          ...   
19086              NaN          7       Oct-49         Pvt Ltd            1   
19097              NaN          7          NaN             NaN            1   
19104              NaN        >20          <10   Public Sector           >4   
19132              NaN          6          NaN             NaN            1   
19152              NaN         10          NaN             NaN            2   

       training_hours  target  
19                  4       0  
65                 82       0  
77                 27       0  
228               108       0  
229                48       0  
...               ...     ...  
19086              45       0  
19097             272       1  
19104              19       1  
19132              98       0  
19152               3       0  

[460 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 460 Rows with missing education level<br>so there is three options:<ol><li>Fill with "Not provided"</li><li>Drop Rows</li><li>Fill with the mode</li></ol></span>

In [344]:
clean_apps['education_level']=apps['education_level'].fillna('Not provided') # I choose to fill with "Not provided"

<span style="font-size: 20px; color: lightblue;">Result: Missing education level filled with "Not provided"</span>

In [345]:
missing_major=apps["major_discipline"].isnull()
apps[missing_major]

enrollee_id      city  city_development_index             gender  \
0                1  city_103                   0.920               Male   
8               11  city_103                   0.920               Male   
19              36   city_10                   0.895               Male   
21              41  city_100                   0.887               Male   
30              53   city_16                   0.910  prefer not to say   
...            ...       ...                     ...                ...   
19132        33338  city_103                   0.920  prefer not to say   
19135        33341   city_75                   0.939               Male   
19136        33342   city_21                   0.624               Male   
19150        33370  city_114                   0.926               Male   
19152        33374   city_61                   0.913               Male   

           relevent_experience enrolled_university education_level  \
0       No relevent experience       no_enrollment     High School   
8      Has relevent experience    Part time course  Primary School   
19      No relevent experience       no_enrollment    Not provided   
21     Has relevent experience       no_enrollment     High School   
30      No relevent experience    Full time course     High School   
...                        ...                 ...             ...   
19132   No relevent experience    Part time course    Not provided   
19135  Has relevent experience    Full time course     High School   
19136   No relevent experience    Full time course     High School   
19150   No relevent experience    Full time course     High School   
19152  Has relevent experience       no_enrollment    Not provided   

      major_discipline experience company_size company_type last_new_job  \
0                  NaN          2          NaN      Pvt Ltd        never   
8                  NaN         11          NaN          NaN            4   
19                 NaN          3          <10      Pvt Ltd            2   
21                 NaN        >20      100-500      Pvt Ltd            1   
30                 NaN          7          NaN      Pvt Ltd        never   
...                ...        ...          ...          ...          ...   
19132              NaN          6          NaN          NaN            1   
19135              NaN          6        50-99      Pvt Ltd            1   
19136              NaN          5          NaN      Pvt Ltd        never   
19150              NaN          6          NaN          NaN            1   
19152              NaN         10          NaN          NaN            2   

       training_hours  target  
0                 150       0  
8                  20       0  
19                  4       0  
21                 47       0  
30                 33       0  
...               ...     ...  
19132              98       0  
19135              28       0  
19136               6       0  
19150             264       0  
19152               3       0  

[2813 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 2813 Missing Rows with missing major discipline</span>

In [346]:
clean_apps["major_discipline"].unique() #shows the unique values in specific feature 
clean_apps['major_discipline']=apps['major_discipline'].fillna('Other') # I choose to fill with "Other"

<span style="font-size: 20px; color: lightblue;">Result: Missing major discipline filled with "No Major"</span>

In [347]:
missing_yoexp=apps["experience"].isnull()
apps[missing_yoexp]

enrollee_id      city  city_development_index             gender  \
766           1435  city_103                   0.920  prefer not to say   
1267          2299   city_21                   0.624  prefer not to say   
1459          2700  city_145                   0.555  prefer not to say   
1508          2778   city_21                   0.624  prefer not to say   
1624          2978   city_21                   0.624  prefer not to say   
...            ...       ...                     ...                ...   
16705        29245  city_102                   0.804               Male   
17283        30213   city_21                   0.624  prefer not to say   
17629        30800   city_16                   0.910               Male   
17954        31348   city_99                   0.915              Other   
18688        32626  city_150                   0.698              Other   

           relevent_experience enrolled_university education_level  \
766     No relevent experience       no_enrollment     High School   
1267   Has relevent experience       no_enrollment        Graduate   
1459    No relevent experience    Full time course        Graduate   
1508   Has relevent experience    Full time course        Graduate   
1624   Has relevent experience    Full time course        Graduate   
...                        ...                 ...             ...   
16705   No relevent experience       no_enrollment        Graduate   
17283   No relevent experience       no_enrollment    Not provided   
17629   No relevent experience    Full time course        Graduate   
17954  Has relevent experience       no_enrollment             Phd   
18688  Has relevent experience       no_enrollment    Not provided   

      major_discipline experience company_size   company_type last_new_job  \
766              Other        NaN          NaN            NaN          NaN   
1267              STEM        NaN       Oct-49            NaN          NaN   
1459              STEM        NaN       Oct-49        Pvt Ltd            4   
1508              STEM        NaN    5000-9999            NaN            1   
1624             Other        NaN       10000+  Public Sector          NaN   
...                ...        ...          ...            ...          ...   
16705       Humanities        NaN          NaN            NaN            1   
17283            Other        NaN          NaN            NaN        never   
17629             STEM        NaN      500-999            NaN            1   
17954             STEM        NaN    1000-4999        Pvt Ltd           >4   
18688            Other        NaN       Oct-49            NaN            1   

       training_hours  target  
766                 4       0  
1267              182       1  
1459               83       1  
1508              214       1  
1624               91       0  
...               ...     ...  
16705              91       1  
17283              44       0  
17629             153       0  
17954              35       0  
18688              51       1  

[65 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 65 Rows with missing years of experience<br>**I could drop rows with no experience<br>**But i'll fill the missings with 0 years</span>

In [348]:
clean_apps['experience']=apps['experience'].fillna(0) # I choose to fill with "0"

<span style="font-size: 20px; color: lightblue;">Result: Missing years of experience filled with "0"</span>

In [349]:
missing_compsz=apps["company_size"].isnull()
apps[missing_compsz]

enrollee_id      city  city_development_index             gender  \
0                1  city_103                   0.920               Male   
4                7  city_136                   0.897               Male   
6                9   city_16                   0.910               Male   
8               11  city_103                   0.920               Male   
13              20  city_114                   0.926               Male   
...            ...       ...                     ...                ...   
19137        33344   city_19                   0.682  prefer not to say   
19145        33360  city_103                   0.920               Male   
19148        33367  city_103                   0.920               Male   
19150        33370  city_114                   0.926               Male   
19152        33374   city_61                   0.913               Male   

           relevent_experience enrolled_university education_level  \
0       No relevent experience       no_enrollment     High School   
4      Has relevent experience    Full time course         Masters   
6      Has relevent experience       no_enrollment        Graduate   
8      Has relevent experience    Part time course  Primary School   
13      No relevent experience    Full time course         Masters   
...                        ...                 ...             ...   
19137  Has relevent experience       no_enrollment        Graduate   
19145  Has relevent experience       no_enrollment         Masters   
19148  Has relevent experience       no_enrollment        Graduate   
19150   No relevent experience    Full time course     High School   
19152  Has relevent experience       no_enrollment    Not provided   

      major_discipline experience company_size company_type last_new_job  \
0                Other          2          NaN      Pvt Ltd        never   
4                 STEM         13          NaN          NaN            1   
6                 STEM        >20          NaN          NaN           >4   
8                Other         11          NaN          NaN            4   
13                STEM          7          NaN          NaN            2   
...                ...        ...          ...          ...          ...   
19137             STEM          6          NaN          NaN            1   
19145             STEM         17          NaN          NaN           >4   
19148             STEM        >20          NaN          NaN           >4   
19150            Other          6          NaN          NaN            1   
19152            Other         10          NaN          NaN            2   

       training_hours  target  
0                 150       0  
4                  18       0  
6                   2       1  
8                  20       0  
13                 76       1  
...               ...     ...  
19137              53       1  
19145              65       1  
19148             220       1  
19150             264       0  
19152               3       0  

[5938 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 5938 Rows with missing company size<br>**I will fill with 0 and we can make the drop decision later on.</span>

In [350]:
clean_apps['company_size']=apps['company_size'].fillna(0) # I choose to fill with 0

<span style="font-size: 20px; color: lightblue;">Result: Missing company size filled with 0</span>

In [351]:
print(clean_apps['company_size'].unique())
test_wrong=clean_apps["company_size"]=='Oct-49'
clean_apps[test_wrong]

[0 '10000+' '50-99' '<10' '100-500' '1000-4999' '5000-9999' 'Oct-49'
 '500-999']


enrollee_id      city  city_development_index             gender  \
11              18   city_23                   0.899               Male   
41              72   city_16                   0.910               Male   
52              87   city_16                   0.910               Male   
54              98   city_16                   0.910  prefer not to say   
56             101  city_136                   0.897               Male   
...            ...       ...                     ...                ...   
19100        33289   city_21                   0.624               Male   
19107        33301   city_21                   0.624               Male   
19108        33302   city_73                   0.754               Male   
19124        33328  city_114                   0.926               Male   
19139        33349   city_21                   0.624               Male   

           relevent_experience enrolled_university education_level  \
11      No relevent experience       no_enrollment        Graduate   
41     Has relevent experience       no_enrollment        Graduate   
52     Has relevent experience       no_enrollment        Graduate   
54      No relevent experience       no_enrollment        Graduate   
56     Has relevent experience       no_enrollment         Masters   
...                        ...                 ...             ...   
19100  Has relevent experience       no_enrollment         Masters   
19107  Has relevent experience       no_enrollment         Masters   
19108  Has relevent experience       no_enrollment        Graduate   
19124  Has relevent experience       no_enrollment        Graduate   
19139   No relevent experience    Part time course        Graduate   

      major_discipline experience company_size         company_type  \
11                STEM        >20       Oct-49              Pvt Ltd   
41     Business Degree          1       Oct-49              Pvt Ltd   
52          Humanities          3       Oct-49              Pvt Ltd   
54                STEM          6       Oct-49              Pvt Ltd   
56                STEM         19       Oct-49       Funded Startup   
...                ...        ...          ...                  ...   
19100             STEM         10       Oct-49              Pvt Ltd   
19107             STEM         14       Oct-49                  NGO   
19108             STEM          6       Oct-49              Pvt Ltd   
19124             STEM         13       Oct-49              Pvt Ltd   
19139             STEM          2       Oct-49  Early Stage Startup   

      last_new_job  training_hours  target  
11              >4              24       0  
41               1               8       0  
52               1              76       0  
54              >4              40       0  
56               3              12       0  
...            ...             ...     ...  
19100            3              18       0  
19107            2             268       1  
19108            1               7       0  
19124            1              97       0  
19139            2              20       0  

[1471 rows x 14 columns]

In [352]:
clean_apps=clean_apps[clean_apps['company_size']!='Oct-49'] #drop rows with "Oct-49"

<span style="font-size: 20px; color: lightblue;">Result: Spotted a repeated wrong value "Oct-49"<br>**So, i'll drop the rows with this value<br>**The main reason of dropping is that i found the majority of rows with "Oct-49" company size, has lack of information in other features</span>

In [353]:
missing_comptp=apps["company_type"].isnull()
apps[missing_comptp]

enrollee_id      city  city_development_index             gender  \
4                7  city_136                   0.897               Male   
6                9   city_16                   0.910               Male   
8               11  city_103                   0.920               Male   
13              20  city_114                   0.926               Male   
14              21   city_73                   0.754               Male   
...            ...       ...                     ...                ...   
19137        33344   city_19                   0.682  prefer not to say   
19145        33360  city_103                   0.920               Male   
19148        33367  city_103                   0.920               Male   
19150        33370  city_114                   0.926               Male   
19152        33374   city_61                   0.913               Male   

           relevent_experience enrolled_university education_level  \
4      Has relevent experience    Full time course         Masters   
6      Has relevent experience       no_enrollment        Graduate   
8      Has relevent experience    Part time course  Primary School   
13      No relevent experience    Full time course         Masters   
14     Has relevent experience    Full time course        Graduate   
...                        ...                 ...             ...   
19137  Has relevent experience       no_enrollment        Graduate   
19145  Has relevent experience       no_enrollment         Masters   
19148  Has relevent experience       no_enrollment        Graduate   
19150   No relevent experience    Full time course     High School   
19152  Has relevent experience       no_enrollment    Not provided   

      major_discipline experience company_size company_type last_new_job  \
4                 STEM         13            0          NaN            1   
6                 STEM        >20            0          NaN           >4   
8                Other         11            0          NaN            4   
13                STEM          7            0          NaN            2   
14                STEM          8            0          NaN            2   
...                ...        ...          ...          ...          ...   
19137             STEM          6            0          NaN            1   
19145             STEM         17            0          NaN           >4   
19148             STEM        >20            0          NaN           >4   
19150            Other          6            0          NaN            1   
19152            Other         10            0          NaN            2   

       training_hours  target  
4                  18       0  
6                   2       1  
8                  20       0  
13                 76       1  
14                 90       0  
...               ...     ...  
19137              53       1  
19145              65       1  
19148             220       1  
19150             264       0  
19152               3       0  

[6140 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 6140 Rows with missing company type</span>

In [354]:
clean_apps.loc[:, 'company_type'] = clean_apps['company_type'].fillna('Other')

<span style="font-size: 20px; color: lightblue;">Result: Missing company type filled with "Other"</span>

In [355]:
missing_lastnewjob=apps["last_new_job"].isnull()
apps[missing_lastnewjob]

enrollee_id      city  city_development_index             gender  \
94             169   city_98                   0.949  prefer not to say   
122            225  city_173                   0.878  prefer not to say   
236            436   city_11                   0.550  prefer not to say   
241            443   city_24                   0.698  prefer not to say   
244            446   city_74                   0.579  prefer not to say   
...            ...       ...                     ...                ...   
18697        32641   city_21                   0.624               Male   
18726        32687   city_42                   0.563  prefer not to say   
18748        32721  city_103                   0.920  prefer not to say   
18797        32798   city_79                   0.698  prefer not to say   
18837        32854   city_10                   0.895  prefer not to say   

           relevent_experience enrolled_university education_level  \
94     Has relevent experience       no_enrollment         Masters   
122    Has relevent experience       no_enrollment         Masters   
236     No relevent experience    Part time course        Graduate   
241     No relevent experience    Full time course     High School   
244     No relevent experience    Full time course     High School   
...                        ...                 ...             ...   
18697   No relevent experience       no_enrollment        Graduate   
18726  Has relevent experience    Full time course             Phd   
18748  Has relevent experience       no_enrollment        Graduate   
18797  Has relevent experience    Full time course         Masters   
18837   No relevent experience    Full time course     High School   

      major_discipline experience company_size    company_type last_new_job  \
94                STEM        >20    1000-4999         Pvt Ltd          NaN   
122               STEM        >20        50-99  Funded Startup          NaN   
236               STEM          4            0             NaN          NaN   
241              Other          1            0             NaN          NaN   
244              Other          1            0             NaN          NaN   
...                ...        ...          ...             ...          ...   
18697            Other          3            0             NaN          NaN   
18726             STEM         <1            0             NaN          NaN   
18748             STEM          7            0   Public Sector          NaN   
18797             STEM          2        50-99         Pvt Ltd          NaN   
18837            Other          8            0             NaN          NaN   

       training_hours  target  
94                 62       0  
122                19       0  
236                 4       0  
241                 6       0  
244               102       1  
...               ...     ...  
18697             100       1  
18726              43       1  
18748              11       0  
18797              95       0  
18837             110       0  

[423 rows x 14 columns]

<span style="font-size: 20px; color: lightblue;">Result: 423 Rows with missing last new job</span>

In [356]:
clean_apps.loc[:,'last_new_job']=clean_apps['last_new_job'].fillna('never')

<span style="font-size: 20px; color: lightblue;">Result: Missing last new job filled with "never"</span>

In [357]:
missing_trainh=apps["training_hours"].isnull()
apps[missing_trainh]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No missing training hours</span>

In [358]:
missing_target=apps["target"].isnull()
apps[missing_target]

Empty DataFrame
Columns: [enrollee_id, city, city_development_index, gender, relevent_experience, enrolled_university, education_level, major_discipline, experience, company_size, company_type, last_new_job, training_hours, target]
Index: []

<span style="font-size: 20px; color: lightblue;">Result: No missing target found<br>**Target is a boolean feature that shows if the enrollee is looking for job change (1) or not (0)</span>

In [359]:
apps=clean_apps.copy()
del clean_apps

<span style="font-size: 20px; color: #FFAAAA;">Closing up by moving the clean data to main datafarme<br>**And deleting the other dataframe for memory optimization</span>

In [360]:
apps.to_csv('Data_CSV/Clean_Data.csv', index=False)

<span style="font-size: 30px; color: #808080; font-weight:bold;">Finally!, saving our work in a new CSV file.</span>